scout/server/blueprints/variant/utils.py

```diff
@@ -615,19 +615,31 @@ def get_filters(variant_obj: dict) -> List[Dict[str, str]]:
     """Return a list with richer format descriptions about filter status,
     if available. Fall back to display the filter status in a "danger" badge
     if it is not known from the VARIANT_FILTERS constant.
+
+    Currently, the controller may be applied repeatedly to the same variant object
+    for redecoration. Check if the filter strings have already been converted
+    to dicts before attemting to convert them.
     """
-    return [
-        (
-            VARIANT_FILTERS[f]
-            if f in VARIANT_FILTERS
-            else {
-                "label": f.replace("_", " ").upper(),
-                "description": f.replace("_", " "),
-                "label_class": "danger",
-            }
-        )
-        for f in map(lambda x: x.lower(), variant_obj["filters"])
-    ]
+    variant_filters = variant_obj["filters"]
+
+    filters: List[Dict[str, str]] = []
+
+    for f in variant_filters:
```

Comment:
This works, but I like the idea of refactoring because when we have to start having these solutions it means we have some convolute code before..

Reply:
Indeed. It would seem natural to pull the only-show-a-variant-once code in from the template to the controller, and then invoke the variant decorator only once per variant.

Reply:
I made an issue so we don't forget: https://github.com/Clinical-Genomics/scout/issues/5078

Reply:
Nice, I saw! 🚀